packages/core/src/IntersectionDetector/IntersectionDetector.js

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useRef } from 'react'
+import { PropTypes } from '@dhis2/prop-types'
+;('') // TODO: https://github.com/jsdoc/jsdoc/issues/1718
+
+/**
+ * @module
+ *
+ * @param {IntersectionDetector.PropTypes} props
+ * @returns {React.Component}
+ */
+export const IntersectionDetector = ({
+    threshold,
+    onVisibilityChange,
+    children,
+    className,
+    rootRef,
+}) => {
+    // use ref to prevent unnecessary re-render
```

Comment:
I think it'd be useful to elaborate slightly here with what you told me on slack (why not state, etc.), so that other devs understand the why.

Reply:
What do you think of this suggestion? I see it was resolved but am not sure what the resolution was.